### PR TITLE
use ust to get stacktrace with line number

### PR DESCRIPTION
**Pull request by PikachuHy**
ust GitHub repo

https://github.com/MisterTea/UniversalStacktrace

MacOS only

```

Backtrace of C++ stack:
[0] 0x1038dd3aa get_stacktrace(unsigned int) (stack_trace.cpp:23)
[1] 0x10392e899 get_crash_report(char const*) (tm_debug.cpp:131)
[2] 0x103ec7602 QTMWidget::resizeEvent(QResizeEvent*) (QTMWidget.cpp:217)
[3] 0x105d606b0 QWidget::event(QEvent*)
[4] 0x105dfaabb QFrame::event(QEvent*)
[5] 0x103eb5e8b QTMScrollView::surfaceEvent(QEvent*) (QTMScrollView.cpp:245)
[6] 0x103eb6029 QTMSurface::event(QEvent*) (QTMScrollView.cpp:43)
[7] 0x105d0dda7 QApplicationPrivate::notify_helper(QObject*, QEvent*)
[8] 0x105d0ed63 QApplication::notify(QObject*, QEvent*)
[9] 0x102f46385 QTMApplication::notify(QObject*, QEvent*) (QTMApplication.hpp:149)
[10] 0x106e5d3ba QCoreApplication::notifyInternal2(QObject*, QEvent*)
[11] 0x105d5d4f2 QWidgetPrivate::setGeometry_sys(int, int, int, int, bool)
[12] 0x105d5d5be QWidget::resize(QSize const&)
[13] 0x105d56b9b QWidget::setMinimumSize(int, int)
[14] 0x105d573b4 QWidget::setMinimumHeight(int)

......
```

### CODE REVIEW

**Comment by PikachuHy** [comment]
rather slow
```
11/19 Test #11: list_test ........................   Passed   11.33 sec
```

File: src/Plugins/UniversalStacktrace/ust/ust.hpp [hunk 1]
```diff
@@ -0,0 +1,510 @@
+#pragma once
```
Comment (darcy-shen) [inline src/Plugins/UniversalStacktrace/ust/ust.hpp:1]: Two necessary info:
+ LICENSE and Author
+ the original repo and the commit id

**Review by darcy-shen — CHANGES_REQUESTED** [review]
Need more info of ust

File: src/Plugins/UniversalStacktrace/ust/ust.hpp (hunk 1 above)
Comment (darcy-shen) [reply src/Plugins/UniversalStacktrace/ust/ust.hpp:1]: could you add these info to a README.md under src/Plugins/UniversalStacktrace

Comment (PikachuHy) [reply src/Plugins/UniversalStacktrace/ust/ust.hpp:1]: sure

**Review by darcy-shen — APPROVED** [review]

**Pull request merged**